rts/Rendering/GL/glHelpers.h

```diff
@@ -0,0 +1,69 @@
+/* This file is part of the Recoil engine (GPL v2 or later), see LICENSE.html */
+
+#pragma once
+
+#include "myGL.h"
+#include <algorithm>
+#include <tuple>
+
+
+// Must pass expectedValuesN to convert from other integer types to GLint
+template<class GLType>
+inline void glGetAny(GLenum paramName, GLType* data, const int expectedValuesN = -1)
+{
+	GLint ints[expectedValuesN];
+	glGetIntegerv(paramName, ints);
+	std::move(ints, ints+expectedValuesN, data);
+}
+template<>
+inline void glGetAny<GLint>(GLenum paramName, GLint* data, const int)
+{
+	glGetIntegerv(paramName, data);
+}
+template<>
+inline void glGetAny<GLboolean>(GLenum paramName, GLboolean* data, const int)
+{
+	glGetBooleanv(paramName, data);
+}
+template<>
+inline void glGetAny<GLfloat>(GLenum paramName, GLfloat* data, const int)
+{
+	glGetFloatv(paramName, data);
+}
+template<>
+inline void glGetAny<GLdouble>(GLenum paramName, GLdouble* data, const int)
+{
+	glGetDoublev(paramName, data);
+}
+
+
+namespace GL
+{
+
+template<class ResultTupleType, GLenum... GLParamNames>
```

Comment:
Seems like your implementation only allows for one or two `GLParamNames`, I think it will be cleaner two versions of this function respectively